### ISSUE TITLE
Metric aggragation is broken for LoggerCollection

### ISSUE THREAD

**Issue by festeh** [PyTorchLightning/pytorch-lightning#1547]
## 🐛 Bug

After changes in #1278 it is now not possible to log testing metrics after traning while using several loggers.
### To Reproduce
Say we want to run a [MINST example](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/basic_examples/cpu_template.py) and also want to add a change - log testing metrics after training. For that we define a Callback

```
class TestCallback(Callback):
    def on_train_end(self, trainer, pl_module):
        # note that it would crash if you don't pass the `pl_module`
        trainer.test(pl_module)
```
and pass it to trainer `callbacks` argument. 

We would also like to use several loggers to track all metrics, say `MLFlowLogger` and `TensorBoardLogger`. For this we create instances of these loggers and pass them into `Trainer` in a list.

### Expected behavior
Testing metrics should be logged - but they don't as there's no final aggregation when our logger is a `LoggerCollection`

### Additional context
In my opinion, the logic in `agg_and_log_metrics`  and `_finalize_agg_metrics` is hard to follow, so I'd be happy if user could choose plain old `log_metrics` which worked nicely.

**Comment by Borda** [comment]
are you using the last master?

**Comment by festeh** [comment]
@Borda thanks! It now works for a single logger. But it seems that problem persists when several loggers are used, so I updated the issue.

**Comment by kevinc13** [comment]
I'm also having this issue with multiple loggers on master. Seems that the test results are printed at the end, but the loggers don't log the test metrics.

**Comment by kevinc13** [comment]
@Borda any update on this issue? I'm still encountering this on the latest release, and it seems related to #1859
